src/org/pahappa/systems/registrationapp/views/UserView.java

```diff
@@ -59,21 +66,125 @@ public void displayMenu() {
+        String username = scanner.nextLine();
+        System.out.println("Enter Date of Birth: mm/dd/yyyy");
+        String dateOfBirth = scanner.nextLine();
+
+        if(firstName.isEmpty() || lastName.isEmpty()|| username.isEmpty() || dateOfBirth.isEmpty()) {
+            System.out.println("All Fields required.");
+        }
+        else {
+
+
+            DateFormat birthDate = new SimpleDateFormat("mm/dd/yyyy");
+            Date date_of_birth = birthDate.parse(dateOfBirth);
+            boolean user_name_present = false;
+            for (User x : UserService.users) {
+                if ((x.getUsername().equals(username))) {
+                    user_name_present = true;
+                }
+            }
+            if (user_name_present) {
+
+                System.out.println("Username exists. Please try again.");
+
+            } else {
+                User user = new User(); // Create a new User object
+                user.setFirstname(firstName);
+                user.setLastname(lastName);
+                user.setUsername(username);
+                user.setDateOfBirth(date_of_birth);
+                UserService.users.add(user);
+                System.out.println("User registered successfully.");
+            }
+
+
+        }
+
+
+
     }
 
     private void displayAllUsers() {
+        if(!UserService.users.isEmpty()) {
+            for (User u : UserService.users) {
+                System.out.println("User name: " + u.getUsername() + " has names " + u.getFirstname() + " " + u.getLastname() + "and date of birth " + u.getDateOfBirth());
+            }
+        }
+        else {
+            System.out.println("There are no users registered.");
+        }
     }
 
     private void getUserOfUsername() {
+        System.out.print("Enter username for the user you wish to search: ");
+
+        for (User u : UserService.users) {
+            if (u.getUsername().equals(scanner.nextLine())) {
+                System.out.println("User name: " + u.getUsername() + " has names " + u.getFirstname() + " " + u.getLastname()+ "and date of birth " + u.getDateOfBirth());
+            }
+            else {
+                System.out.println("Username does not match any user.");
+            }
+        }
+
     }
 
-    private void updateUserOfUsername() {
+    private void updateUserOfUsername() throws ParseException {
+        System.out.print("Enter username for the user you wish to update ");
+        String username = scanner.nextLine();
+        for (User u : UserService.users) {
+            if (u.getUsername().equals(username)) {
+                System.out.print("Enter new first name: "); // Prompt user for input
+                String firstName = scanner.nextLine();
+                System.out.print("Enter new last name: ");
+                String lastName = scanner.nextLine();
+                System.out.print("Enter new  date of birth: ");
+                String dateOfBirth = scanner.nextLine();
```

Comment:
On updating user details, notify the user of the date format you want them to provide